src/fragments/lib/auth/flutter/signin_with_custom_flow/10_cli_setup.mdx

```diff
@@ -0,0 +1,182 @@
+
+<Callout>
+  This challenge is expecting a simple, arbitrary value for each authentication
+  attempt. This is not a secure authentication mechanism and is included for
+  demonstration purposes only. For production-ready applications, you would use
+  this challenge to create a more secure authentication mechanism, such as using
+  SES to deliver a randomly generated code.
+</Callout>
+
+Save and close the file, then switch back to the terminal and follow the instructions:
+
+```
+? Press enter to continue
+    `Hit Enter`
+
+? Do you want to edit your boilerplate-define-challenge function now?
+    `Yes`
+? Please edit the file in your editor: <local file path>/src/boilerplate-define-challenge.js
+```
+
+The boilerplate for Define Auth Challenge opens in your favorite code editor. Enter the following code to this file:
+
+```js
+exports.handler = async function(event) {
+  if (
+    event.request.session.length == 1 &&
+    event.request.session[0].challengeName == 'SRP_A'
+  ) {
+    event.response.issueTokens = false;
+    event.response.failAuthentication = false;
+    event.response.challengeName = 'CUSTOM_CHALLENGE';
+  } else if (
+    event.request.session.length == 2 &&
+    event.request.session[1].challengeName == 'CUSTOM_CHALLENGE' &&
+    event.request.session[1].challengeResult == true
+  ) {
+    event.response.issueTokens = true;
+    event.response.failAuthentication = false;
+    event.response.challengeName = 'CUSTOM_CHALLENGE';
+  } else {
+    event.response.issueTokens = false;
+    event.response.failAuthentication = true;
+  }
+};
+```
+
+Note that each of these if/else blocks are referencing `event.request.session`. The session is a JS array in which each index represents a step in the custom authentication flow.
+
+The Amplify Auth library always starts with an SRP_A flow, so in the code above, we bypass SRP_A and return `CUSTOM_CHALLENGE` in the first step. In the second step, if `CUSTOM_CHALLENGE` returns with `challengeResult == true` we recognize the custom auth challenge is successful, and tell Cognito to issue tokens. In the last `else` block we tell Cognito to fail the authentication flow.
+
+Save and close the file, then switch back to the terminal and follow the instructions:
+
+```
+? Press enter to continue
+    `Hit Enter`
+
+? Do you want to edit your boilerplate-verify function now?
+    `Yes`
+? Please edit the file in your editor: <local file path>/src/boilerplate-verify.js
+```
+
+The boilerplate for Verify Auth Challenge opens in your favorite code editor. Enter the following code to this file:
+
+```js
+function verifyAuthChallengeResponse(event) {
+  if (
+    event.request.privateChallengeParameters.answer ===
+    event.request.challengeAnswer
+  ) {
+    event.response.answerCorrect = true;
+  } else {
+    event.response.answerCorrect = false;
+  }
+}
+
```

Comment:
Can we add comments to these code snippets? Specifically the If statement here

Reply:
I'm going to leave these uncommented for now, because I think we explain what's happening in the text.